include/itkBlockMatchingNormalizedCrossCorrelationNeighborhoodIteratorMetricImageFilter.hxx

```diff
@@ -136,24 +142,60 @@ NormalizedCrossCorrelationNeighborhoodIteratorMetricImageFilter< TFixedImage, TM
     MetricIteratorType metricIt( metricPtr, metricRegion );
 
     MetricConstIteratorType denomIt( denom, *fit );
-    NeighborhoodIteratorType movingNeighborIt( radius, movingMinusMean, *fit );
-    movingNeighborIt.OverrideBoundaryCondition( &boundaryCondition );
+
 
     // for every pixel in the output
-    for( metricIt.GoToBegin(), denomIt.GoToBegin(), movingNeighborIt.GoToBegin();
+    //NeighborhoodIteratorType movingNeighborIt( radius, movingMinusMean, *fit );
+    //movingNeighborIt.OverrideBoundaryCondition( &boundaryCondition );
+    //for( metricIt.GoToBegin(), denomIt.GoToBegin(), movingNeighborIt.GoToBegin();
+         //!metricIt.IsAtEnd();
+         //++metricIt, ++denomIt, ++movingNeighborIt )
+      //{
+      //if( !(denomIt.Get() == NumericTraits< MetricImagePixelType >::Zero) )
+        //{
+        //const MetricImagePixelType normXcorr = innerProduct( movingNeighborIt, fixedKernelOperator ) / denomIt.Get();
+
+        //// Why does this happen?  Bug?  Funky floating point behavior?
+        //if( normXcorr < negativeOne )
+          //{
+          //metricIt.Set( negativeOne );
+          //}
+        //else if ( normXcorr > positiveOne )
+          //{
+          //metricIt.Set( positiveOne );
+          //}
+        //else
+          //{
+          //metricIt.Set( normXcorr );
+          //}
+        //}
+      //}
+    for( metricIt.GoToBegin(), denomIt.GoToBegin();
          !metricIt.IsAtEnd();
-         ++metricIt, ++denomIt, ++movingNeighborIt )
+         ++metricIt, ++denomIt)
       {
       if( !(denomIt.Get() == NumericTraits< MetricImagePixelType >::Zero) )
         {
-        normXcorr = innerProduct( movingNeighborIt, fixedKernelOperator ) / denomIt.Get();
+        using RangeType = Experimental::ShapedImageNeighborhoodRange<const MetricImageType,
+          Experimental::ConstantBoundaryImageNeighborhoodPixelAccessPolicy<const MetricImageType> >;
+        const RangeType movingRange{ *movingMinusMean, denomIt.GetIndex(), offsets };
+        const RangeType kernelRange{ *fixedMinusMean, denomIt.GetIndex(), offsets };
```

Comment:
Is it really intended that the kernel is set on a new location (denomIt.GetIndex()) with every iteration?